assets/js/util/convert-time.js

```diff
@@ -58,7 +58,9 @@ export const convertSecondsToArray = ( seconds ) => {
 export const convertDateStringToUNIXTimestamp = ( dateStringValue ) => {
 	const unixTimestamp =
 		dateStringValue && ! Number.isInteger( dateStringValue )
-			? new Date( dateStringValue ).getTime()
+			? // Valid use of `new Date()` with an argument.
+			  // eslint-disable-next-line sitekit/no-direct-date
+			  new Date( dateStringValue ).getTime()
```

Comment:
This is valid if it uses a timestamp as shown in the docblock (complete with timezone), however this would be incorrect for one of our `YYYY-MM-DD` date strings as noted in the description of the issue.

Reply:
Oh, right, I missed the ternary showing that this _would_ be a string in this case 😓 Thanks 👍🏻

Reply:
Oh, actually, this is only used in one place and is designed to accept strings:

https://github.com/google/site-kit-wp/blob/c7de71d253f7b502e24db9950dbd33cc7854c299/assets/js/modules/analytics-4/datastore/base.js#L78-L82

Linting isn't the right way to protect against invalid arguments here, it would need to be an `invariant` on the length of the datetime string.

Reply:
Agreed, SGTM